src/components/genreSelect/genreSelect.test.js

```diff
@@ -1,26 +1,36 @@
-import '@testing-library/jest-dom'
-
-import React from 'react'
-import { fireEvent, render, screen } from '@testing-library/react'
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
 import GenreSelect from './genreSelect';
 
-const genreSelectCB = jest.fn();
-const setup = (genres, selectedGenre) => {
-    const utils = render(<GenreSelect
-        genres={genres}
-        selectedGenre={selectedGenre}
-        onChange={genreSelectCB}></GenreSelect>);
-    return { ...utils };
-}
-describe('Test Genre Select Component', () => {
-    test('check if the genre select is rendered with initial values', () => {
-        setup(["Drama", "Horror"], "Drama");
-        expect(screen.getByText("Drama")).not.toBeNull();
-    })
-
-    test('check if the selected genre is highlighted', () => {
-        setup(["Drama", "Horror"], "Drama");
-        fireEvent.click(screen.getByText("Horror"));
-        expect(genreSelectCB).toBeCalledWith("Horror");
-    })
-})
+const mockGenres = ['ACTION', 'DRAMA', 'COMEDY'];
+const mockSelectedGenre = 'DRAMA';
+const mockOnChange = jest.fn();
+
+describe('GenreSelect Component', () => {
+    it('renders GenreList with provided genres and selected genre', () => {
+        const { getByText } = render(<GenreSelect genres={mockGenres} selectedGenre={mockSelectedGenre} onChange={mockOnChange} />);
+
+        mockGenres.forEach((genre) => {
+            expect(getByText(genre)).toBeInTheDocument();
+        });
+
+        expect(getByText(mockSelectedGenre)).toHaveClass('selected-genre');
+    });
+
+    it('calls onChange prop when a genre is selected', () => {
+        const { getByText } = render(<GenreSelect genres={mockGenres} selectedGenre={mockSelectedGenre} onChange={mockOnChange} />);
+
+        fireEvent.click(getByText('COMEDY'));
+
+        expect(mockOnChange).toHaveBeenCalledWith('COMEDY');
+    });
+
+    it('updates selected genre state when a genre is selected', () => {
+        const { getByText } = render(<GenreSelect genres={mockGenres} selectedGenre={mockSelectedGenre} onChange={mockOnChange} />);
+
+        fireEvent.click(getByText('ACTION'));
+
+        expect(getByText('ACTION')).toHaveClass('selected-genre');
+        expect(mockOnChange).toHaveBeenCalledWith('ACTION');
+    });
+});
```

Comment:
Move test file to __test__ folder, Please follow the same for other test file as well.

Reply:
Since I have single unit test file for each source file. Is it ok to keep them alongside. Otherwise it might become cumbersome to manage:
[https://stackoverflow.com/a/62765143/3019006](https://stackoverflow.com/a/62765143/3019006)